### PR TITLE
Split `zcash_keys` crate out from `zcash_client_backend`

### PR DESCRIPTION
**Pull request by nuttycom**
This change makes it easier for third parties to make use of the Unified key infrastructure without incurring a dependency upon the rest of the `zcash_client_backend` interfaces.

### CODE REVIEW

File: zcash_keys/LICENSE-MIT [hunk 1]
```diff
@@ -0,0 +1,21 @@
+The MIT License (MIT)
+
+Copyright (c) 2017-2019 Electric Coin Company
```
Comment (daira) [inline zcash_keys/LICENSE-MIT:3]:
```suggestion
Copyright (c) 2017-2024 Electric Coin Company
```

**Review by daira — DISMISSED** [review]
utACK modulo adding changelog and fixing copyright date (non-blocking).

**Review by daira — APPROVED** [review]
re-utACK; changelog entries reviewed.

File: zcash_keys/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,91 @@
+[package]
+name = "zcash_keys"
+description = "Zcash key and address management"
+version = "0.1.0"
```
Comment (str4d) [inline zcash_keys/Cargo.toml:4]:
```suggestion
version = "0.0.0"
```
until this is in a state we want to publish.

**Review by str4d — APPROVED** [review]
Concept ACK, but NACK on tagging the tip of this branch for publishing; we need at least one more PR to go over the API.

**Review by str4d — APPROVED** [review]
utACK 0ca955336dca7ef82d667a2e69fcc562143079d4.

**Pull request merged**